### ISSUE TITLE
Add support for UML package diagrams

### ISSUE THREAD

**Issue by clue** [clue/graph-uml#4]
Useful for depicting dependencies between components (composer, packagist).

Could also be useful for: https://github.com/clue/graph-composer

http://en.wikipedia.org/wiki/Package_diagram
http://www.visual-paradigm.com/product/vpuml/tutorials/packagediagram.jsp

**Comment by clemens-tolboom** [comment]
Reading about shapes on http://www.graphviz.org/content/node-shapes we 'only' have to support for subgraphs.

Ie http://twu.seanho.com/11spr/cmpt166/uml/packagediagram.gif

**Comment by clue** [comment]
> support for subgraphs

Indeed, we need to support subgraphs (GraphViz clusters) in clue/graph for subpackages in UML package diagrams.

Additionally, we need to expose an interface in order actually be able to construct UML package diagrams, packages and perhaps dedicated edge types.

**Comment by clemens-tolboom** [comment]
subgraphs issue is https://github.com/clue/graph/pull/36

@clue please sketch that interface

**Comment by clemens-tolboom** [comment]
Another simple example http://trac.edgewall.org/attachment/wiki/TracDev/ComponentArchitecture/comparch.png

I still failed to generate one by hand through graphviz :(
